packages/valory/skills/dynamic_nft_abci/handlers.py

```diff
@@ -84,6 +85,23 @@ class HttpHandler(BaseHttpHandler):
 
     def setup(self) -> None:
         """Implement the setup."""
+        uri_base_hostname = urlparse(self.context.params.token_uri_base).hostname
+
+        # Route regexes
+        hostname_regex = rf".*({uri_base_hostname}|localhost|127.0.0.1)(:\d+)?"
+        self.handler_url_regex = rf"{hostname_regex}\/.*"
+        metadata_url_regex = rf"{hostname_regex}\/\d+"
+        health_url_regex = rf"{hostname_regex}\/healthcheck"
+
+        # Routes
+        self.routes = {
+            (HttpMethod.GET.value, HttpMethod.HEAD.value): [
+                (metadata_url_regex, self._handle_get_metadata),
+                (health_url_regex, self._handle_get_health),
+            ],
+        }
+
+        self.json_content_header = "Content-Type: application/json\n"
```

Comment:
We build a relationship between url regexes and their corresponding handler method. This was inspired by Django's url dispatcher. As they explain here, it is not recommended to have a dict for this, but iterate over the routes.